# **=========================== DATASET ===========================**

In [2]:
import pandas as pd

In [3]:
file = "/content/drive/MyDrive/sisfor/Messy_Employee_dataset.csv"
df = pd.read_csv(file)

df

Employee_ID First_Name Last_Name   Age   Department_Region    Status  \
0        EMP1000        Bob     Davis  25.0   DevOps-California    Active   
1        EMP1001        Bob     Brown   NaN       Finance-Texas    Active   
2        EMP1002      Alice     Jones   NaN        Admin-Nevada   Pending   
3        EMP1003        Eva     Davis  25.0        Admin-Nevada  Inactive   
4        EMP1004      Frank  Williams  25.0  Cloud Tech-Florida    Active   
...          ...        ...       ...   ...                 ...       ...   
1015     EMP2015      David    Miller  30.0       HR-California    Active   
1016     EMP2016      David   Johnson  30.0    Cloud Tech-Texas  Inactive   
1017     EMP2017    Charlie  Williams  40.0    Finance-New York    Active   
1018     EMP2018      Alice    Garcia  30.0          HR-Florida  Inactive   
1019     EMP2019      Heidi     Jones  30.0     DevOps-Illinois   Pending   

       Join_Date     Salary                         Email       Phone  \
0       4/2/2021   59767.65         bob.davis@example.com -1651623197   
1      7/10/2020   65304.66         bob.brown@example.com -1898471390   
2      12/7/2023   88145.90       alice.jones@example.com -5596363211   
3     11/27/2021   69450.99         eva.davis@example.com -3476490784   
4       1/5/2022  109324.61    frank.williams@example.com -1586734256   
...          ...        ...                           ...         ...   
1015   8/19/2023        NaN      david.miller@example.com -3546212759   
1016   11/7/2021  100215.06     david.johnson@example.com -2508261122   
1017   10/4/2023  114587.11  charlie.williams@example.com -1261632487   
1018  12/16/2024   71318.79      alice.garcia@example.com -8995729892   
1019   2/22/2021   77764.24       heidi.jones@example.com -7629745492   

     Performance_Score  Remote_Work  
0              Average         True  
1            Excellent         True  
2                 Good         True  
3                 Good         True  
4                 Poor        False  
...                ...          ...  
1015              Good         True  
1016              Good         True  
1017           Average        False  
1018              Good         True  
1019           Average        False  

[1020 rows x 12 columns]

# **=================== BUSINESS UNDERSTANDING ===================**

### **--- BUSINESS OBJECTIVE ---**

Dataset ini merupakan dataset dengan 1020 baris dan 12 kolom. Dataset ini berisi banyak data seputar karyawan seperti ID, nama, departemen, usia, gaji, tanggal bergabung, performa, dan lainnya — tetapi ada banyak masalah seperti duplikasi, format yang tidak konsisten, dan nilai yang hilang.

Dataset ini akan dijadikan sebagai objek utama dari proyek mengubah data karyawan yang “berantakan” menjadi informasi yang rapi dan dapat dipakai untuk pengambilan keputusan bisnis HR.

### **--- ASSESS SITUATION ---**

Saat ini dataset memiliki beberapa masalah data yang umum terjadi, antara lain:
Data duplikat atau redundant
  * Nilai kosong / missing values
  * Format tanggal tidak konsisten
  * Kolom gabungan yang kurang praktis
  * Kesalahan format nilai

Kondisi ini membuat data belum siap langsung dipakai untuk analisis atau model prediktif karena kualitasnya rendah sehingga hasil analisis bisa salah atau tidak akurat.

### **--- ANALYTIC GOALS & PROJECT PLAN ---**

Tujuan analitik dataset ini adalah untuk mengolah data karyawan yang masih berantakan menjadi data yang bersih, konsisten, dan siap dianalisis, sehingga bisa menghasilkan insight yang berguna bagi pengambilan keputusan, khususnya di bidang sumber daya manusia.

Secara lebih rinci, proses analitik diarahkan untuk memastikan bahwa setiap informasi penting seperti usia, departemen, gaji, performa, dan masa kerja karyawan dapat dibaca dengan benar dan mencerminkan kondisi sebenarnya yang bertujuan untuk menggali pola dan hubungan antar data karyawan, misalnya hubungan antara gaji dengan performa, perbedaan karakteristik karyawan antar departemen atau region, serta distribusi usia dan lama bekerja. Pola-pola ini dapat membantu organisasi memahami kondisi internal karyawan secara lebih menyeluruh.

Hasil akhir dari proses analitik diharapkan dapat menjadi dasar pembuatan laporan, visualisasi, maupun analisis lanjutan, seperti evaluasi kinerja karyawan, perencanaan pengembangan SDM, atau persiapan data untuk model prediksi di masa depan. Dengan demikian, data tidak hanya bersih secara teknis, tetapi juga bernilai secara bisnis.